src/lint/validators.nim

```diff
@@ -34,14 +34,27 @@ proc hasValidRuneLength(s, key, path: string; maxLen: int): bool =
                 $maxLen & " characters"
       result.setFalseAndPrint(msg, path)
 
+const
+  emptySetOfStrings = initHashSet[string](0)
+
 proc checkString*(data: JsonNode; key, path: string; isRequired = true,
-                  maxLen = int.high): bool =
+                  allowed = emptySetOfStrings, maxLen = int.high): bool =
   result = true
   if data.hasKey(key):
     case data[key].kind
     of JString:
       let s = data[key].getStr()
-      if s.len > 0:
+      if allowed.len > 0:
+        if s notin allowed:
+          let msgStart = "The value of `" & key & "` is `" & s &
+                          "`, but it must be one of "
```

Comment:
Currently this is losing some context - we'll need to refactor later to regain an error message like
> "The value of `online_editor.indent_style` ..."

In general the `context` handling should be tidied up everywhere.

Reply:
> In general the context handling should be tidied up everywhere.

Agreed. I'm fine with doing that later.